### ISSUE TITLE
How to call set adduser?

### ISSUE THREAD

**Issue by gitbulb** [freenas/iocage-plugin-quassel#1]
While trying to create a plugin, based on this template, I can't figure out how you would need to call upon the adduser method with the required parameters.

I can only seem to call it like:
```
iocage set -P adduser.requiredargs.username=quser adduser.requiredargs.password=pass adduser.requiredargs.fullname=fullname jailname
```
But this seems to result in 3 (separate) calls to adduser, where `$*` is:
```
adduser requiredargs username quser
```
followed by a separate call to adduser, using
```
aduser requiredargs password pass
```
etc..

So user, password and fullname are not provided as list of parameters, but as 3 individual calls to adduser, or at least in the way I figured out iocage 😉 .

Please help @skarekrow or @kmoore134 ?

**Comment by skarekrow** [comment]
afaik adduser is a stub, @kmoore134 you wrote this, do you know?

**Comment by gitbulb** [comment]
The question was more on what the syntax is for calling upon such a property (method), but I think I figured it out. So for the (example) of adduser it would be:
```
iocage set -P "adduser=aUsername,aPassword,A Full Name"
```

This works on the plugin I made ([OpenVPN](https://github.com/gitbulb/iocage-plugin-openvpn)).

Not sure if this is correct?

I's a bit cumbersome because of the placing of quotes, and if you have more then one optional parameters, say 3 and you want to supply the second optional argument, you are required to supply the first optional parameter as well; iocage will split on `=,.` and call the setoptions program/script with that array. So order is crucial, also for the options as there is only a n-th parameter without any assignment to a specific named optional parameter.

But again, based on this I implemented this in the OpenVPN plugin. Now see if it gets accepted 😃 .
